### PR TITLE
Refreshing Tags List: Part III

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix:
In this PR we're fixing a bug that caused the Tag Name to become invisible, on edition.

### Details:
- `NSColor.controlTextColor` is an AppKit provided color, meant for Text onScreen
- Such color has a low level behavior: NSTextField's adjusts automatically, based on the background color
- Using a custom NSColor means loosing the automatic behavior, which ensures the text won't become invisible.

In this PR we're introducing a new **NSTextField** subclass (simply **SPTextField**), capable of automatically switching to a different NSColor, whenever it becomes the first responder.

Nope. There was no standard way to do this 😭 

@aerych last one of the "Tags List" series! yay!!
(Thank you!!!)

Ref. #458

### Test
1. Launch Simplenote
2. Click over any of your tags in the Tags List

- [x] Verify the tag name is visible, on selection
- [x] Verify that the tag is readable after right clicking on it, and selecting **rename**
- [x] Verify that right clicking over any other tag name renders a Menu, and does not cause the field to enter edition mode immediately
- [x] Please repeat for Light / Dark Modes

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/TextField.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+//         within `acceptsFirstResponder`.
+//
+protocol TextFieldDelegate: NSTextFieldDelegate {
+    func controlAcceptsFirstResponder(_ control: NSControl) -> Bool
+}
+
+
+// MARK: - TextField
+//
+class TextField: NSTextField {
```
Comment (aerych) [inline Simplenote/TextField.swift:15]: SwiftUI has a [TextField](https://developer.apple.com/documentation/swiftui/textfield) struct. I know its scoped by framework, but would it be a best practice to use a different name?

File: Simplenote/TextField.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    var activeTextColor: NSColor? {
+        if isEditing {
+            return textEditionColor
+        }
+
+        return isSelected ? textSelectionColor : textRegularColor
+    }
+
+    var innerDelegate: TextFieldDelegate? {
```
Comment (aerych) [inline Simplenote/TextField.swift:94]: This is only used once outside its definition. Are their future plans for it or would it make sense to drop the definition and just try casting the delegate where its used?

Comment (jleandroperez) [reply Simplenote/TextField.swift:94]: Great call sir!! Always think big, but start small.

**Comment by jleandroperez** [comment]
- Renamed **TextField** > **SPTextField** to avoid a future SwiftUI collision
- Fixed a rename regression
- Dropped the `innerDelegate` property

@aerych Thank you **SO MUCH!!!**. Ready for another peek!

**Comment by aerych** [comment]
Looks great @jleandroperez :)  I retested all the things.  :shipit:

**Comment by jleandroperez** [comment]
Thank you sir!!

**Pull request merged**